src/Data/ProgrammesDb/EntityRepository/ImageRepository.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository;
+
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Query;
+
+class ImageRepository extends EntityRepository
+{
+    public function findByPid(string $pid): ?array
+    {
+        $qb = parent::createQueryBuilder('image')
```

Comment:
Is this the simplest that this method can get? I feel this is so generic that it can be refactored even more.

Reply:
It is very rare that we need just the entity without any joins - most of the time we want a FindByPidFull that joins to a set of fields that differs on a per entity basis (e.g. CoreEntity joins to Image & Masterbrand, Contribution join to Contributor etc).

I don't think it's worth creating an abstract thing for this, as 90% of the time it is too constrained to be useful.